R/tar_renv.R

```diff
@@ -0,0 +1,56 @@
+#' @title Setup package dependencies for compatibility with `{renv}`
+#' @export
+#' @description Write package dependencies to a file named `_packages.R` in
+#'   the root project directory. See `Details` for a more extensive
+#'   explanation.
+#' @details This function gets called for its side-effect, which writes
+#'   package dependencies provided to the `packages` argument to
+#'   `_packages.R`. `tar_renv` is unable to detect packages declared after it
+#'   has been called in the script (e.g. target-specific packages declared
+#'   within \code{\link{tar_pipeline}}); these packages need to be
+#'   passed to `tar_renv` manually. The generated file should \strong{not} be
+#'   edited by hand and will be overwritten each time `tar_renv` runs.
+#'
+#'   By explicitly writing packages to a file, `{renv}` is able to crawl the
+#'   file to identify package dependencies with `renv::dependencies()`.
+#'   Calling `renv::init()` after `tar_renv` has ran will set up a
+#'   project-local `R` library. This allows your `targets` pipeline to have
+#'   its own self-contained `R` library separate from your standard `R`
+#'   library. See [`{renv}`](https://rstudio.github.io/renv/index.html) for
+#'   more information.
+#'
+#'   During interactive project development, you may find it useful to run
+#'   `source("_packages.R")` to quickly load packages. However, this file
+#'   primarily exists to provide native compatibility with `{renv}`.
+#' @inheritParams tar_option_set
+#' @param ask Logical, whether to ask before writing if `_packages.R`
+#'   already exists. If `NULL`, defaults to `Sys.getenv("TAR_ASK")`.
+#'   (Set to `"true"` or `"false"` with `Sys.setenv()`).
+#'   If `ask` and the `TAR_ASK` environment variable are both
+#'   indeterminate, defaults to `interactive()`.
+#' @seealso [Introduction to `{renv}`](https://rstudio.github.io/renv/articles/renv.html)
+#' @md
+#' @return Nothing, invisibly.
+#' @examples
+#' tar_dir({
+#' tar_script({
+#'   pkgs <- c("tidyverse", "qs")
+#'   tar_option_set(packages = pkgs)
+#'   tar_renv()
+#'   tar_pipeline()
+#' })
+#' tar_make()
+#' readLines("_packages.R")
+#' })
+tar_renv <- function(packages = targets::tar_option_get("packages"),
+                     ask = NULL) {
+  if (!tar_should_overwrite(ask, "_packages.R")) {
+    return()
```

Comment:
Let's stick a `#nocov` comment next to `return()` and add an interactive semi-automated test at `tests/interactive/test-tar_renv.R` similar to `tests/interactive/test-tar_script.R`.

Also, let's `return(invisible())`.